provider/datagen/src/source.rs

```diff
@@ -110,44 +119,55 @@ impl SourceData {
         }
-        self.with_icuexport(
-            cached_path::CacheBuilder::new().freshness_lifetime(u64::MAX).build().and_then(|cache| cache
-                .cached_path(
-                    &format!("https://github.com/unicode-org/icu/releases/download/{}/icuexportdata_{}.zip", tag, tag.replace('/', "-"))
-            )).map_err(|e| DataError::custom("Download").with_display_context(&e))?)
+        let resource = format!(
+            "https://github.com/unicode-org/icu/releases/download/{}/icuexportdata_{}.zip",
+            tag,
+            tag.replace('/', "-")
+        );
+        Ok(Self {
+            icuexport_paths: Some(Arc::new(SerdeCache::new(AbstractFs::new_lazy(Box::new(
+                move || download(&resource),
+            ))))),
+            ..self
+        })
     }
 
     /// Adds CLDR data to this `DataSource`. This data will be downloaded from the `latest` GitHub tag.
     pub fn with_cldr_latest(self, locale_subset: CldrLocaleSubset) -> Result<Self, DataError> {
-        let response = reqwest::blocking::Client::new()
-            .head("https://github.com/unicode-org/cldr-json/releases/latest")
-            .send()
-            .map_err(|e| DataError::custom("reqwest error").with_display_context(&e))?;
-        self.with_cldr_for_tag(
-            response
-                .url()
-                .path()
-                .split('/')
-                .next_back()
-                .expect("split is non-empty"),
-            locale_subset,
-        )
+        let path = Box::new(move || {
+            let tag = follow_redirect_and_get_last_element(
+                "https://github.com/unicode-org/cldr-json/releases/latest",
+            )?;
+            download(&format!(
+                "https://github.com/unicode-org/cldr-json/releases/download/{}/cldr-{}-json-{}.zip",
+                tag, tag, locale_subset
+            ))
+        });
+        Ok(Self {
+            cldr_paths: Some(Arc::new(CldrCache {
+                cache: SerdeCache::new(AbstractFs::new_lazy(path)),
+                locale_subset,
+            })),
+            ..self
+        })
     }
 
     /// Adds ICU export data to this `DataSource`. This data will be downloaded from the `latest` GitHub tag.
     pub fn with_icuexport_latest(self) -> Result<Self, DataError> {
-        let response = reqwest::blocking::Client::new()
-            .head("https://github.com/unicode-org/icu/releases/latest")
-            .send()
-            .map_err(|e| DataError::custom("reqwest error").with_display_context(&e))?;
-        self.with_icuexport_for_tag(
-            response
-                .url()
-                .path()
-                .split('/')
-                .next_back()
-                .expect("split is non-empty"),
-        )
+        let path = Box::new(move || {
+            let tag = follow_redirect_and_get_last_element(
+                "https://github.com/unicode-org/icu/releases/latest",
+            )?;
```

Comment:
Issue: similar to above. Since we control `icu4exportdata.zip`, we can take more steps to make sure it keeps working for old datagen versions, but why impose that burden on us? People should just upgrade their icu4x-datagen to get the latest and greatest behavior.